### PR TITLE
GA Migration and Sharding Precommit Java IOs

### PR DESCRIPTION
**Pull request by benWize**
As part of the migration of Precommit and Postcommit Jobs from Jenkins to GA in self-hosted runners, this PR contains:

- Migrated and sharded workflow [job-precommit-java-mqtt-io.yml](https://github.com/benWize/beam/blob/88c5ba3e8649fd0f1661113c4ac6447fcb836d50/.github/workflows/job-precommit-java-mqtt-io.yml)
- Migrated and sharded workflow [job-precommit-java-parquet-io.yml](https://github.com/benWize/beam/blob/88c5ba3e8649fd0f1661113c4ac6447fcb836d50/.github/workflows/job-precommit-java-parquet-io.yml)
- Migrated and sharded workflow [job-precommit-java-pulsar-io.ymll](https://github.com/benWize/beam/blob/88c5ba3e8649fd0f1661113c4ac6447fcb836d50/.github/workflows/job-precommit-java-pulsar-io.yml)
- Migrated and sharded workflow [job-precommit-java-rabbitmq-io.yml](https://github.com/benWize/beam/blob/88c5ba3e8649fd0f1661113c4ac6447fcb836d50/.github/workflows/job-precommit-java-rabbitmq-io.yml)
- Migrated and sharded workflow [job-precommit-java-redis-io.yml](https://github.com/benWize/beam/blob/88c5ba3e8649fd0f1661113c4ac6447fcb836d50/.github/workflows/job-precommit-java-redis-io.yml)

The migrated workflows were added to [CI.md](https://github.com/benWize/beam/blob/88c5ba3e8649fd0f1661113c4ac6447fcb836d50/CI.md)

**DO NOT MERGE** until the effort to use **self-hosted runners** is completed https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by benWize** [comment]
Successful executions on self-hosted runners:

- MQTT: https://github.com/benWize/beam/actions/runs/3038852341
- Parquet: https://github.com/benWize/beam/actions/runs/3038852339
- Pulsar: https://github.com/benWize/beam/actions/runs/3038852331
- Redis IO: https://github.com/benWize/beam/actions/runs/3038852328
- RabbitMQ: https://github.com/benWize/beam/actions/runs/3038852327

**Comment by fernando-wizeline** [comment]
LGTM!

File: .github/workflows/job-precommit-java-mqtt-io.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# This workflow will run MQTT IO Unit Tests in Java SDK
+
+name: Java MQTT IO
```
Comment (damccorm) [inline .github/workflows/job-precommit-java-mqtt-io.yml:20]: Should these workflows be removed from Jenkins?

File: CI.md [hunk 2]
```diff
@@ -125,6 +125,33 @@ Service Account shall have following permissions ([IAM roles](https://cloud.goog
 | Java Wordcount Direct Runner | Runs Java WordCount example with Direct Runner.                                               | Yes              | Yes                   | Yes           | -                        |
 | Java Wordcount Dataflow      | Runs Java WordCount example with DataFlow Runner.                                             | -                | Yes                   | Yes           | Yes                      |
 
+### PreCommit Workflows
+
+#### Java MQTT IO - [job-precommit-java-mqtt-io.yml](.github/workflows/job-precommit-java-mqtt-io.yml)
```
Comment (damccorm) [inline CI.md:130]: Could we remove section headers and make this a single table?

Comment (benWize) [reply CI.md]: Done!

File: .github/workflows/job-precommit-java-mqtt-io.yml (hunk 1 above)
Comment (benWize) [reply .github/workflows/job-precommit-java-mqtt-io.yml:20]: These workflows are part of the sharding of the Java Precommit workflow, some of the modules will be placed in a new java precommit workflow in https://github.com/apache/beam/pull/23138, while some others will remain in the current Jenkins job, mainly due to the lack of support for some reports.

**Review by damccorm — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**